LanguageFeatures/Inline-classes/static_analysis_inline_class_A12_t05.dart

```diff
@@ -0,0 +1,63 @@
+// Copyright (c) 2023, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+/// @assertion Let DV be an inline class declaration named Inline with type
+/// parameters X1 extends B1, .. Xs extends Bs. Assume that DV declares a final
+/// instance variable with name id and type R.
+///
+/// We say that the declared representation type of Inline is R, and the
+/// instantiated representation type corresponding to Inline<T1,.. Ts> is
+/// [T1/X1, .. Ts/Xs]R.
+/// ...
+/// Let V be an inline type of the form Inline<T1, .. Ts>, and let R be the
+/// corresponding instantiated representation type. If R is non-nullable then V
+/// is a proper subtype of Object, and V is non-nullable. Otherwise, V is a
+/// proper subtype of Object?, and V is potentially nullable.
+///
+/// @description Checks that it is a compile-time error to have an instance
+/// variable whose type is an inline type, and then relying on implicit
+/// initialization to null.
+/// @author sgrekhov22@gmail.com
+
+// SharedOptions=--enable-experiment=inline-class
+
+inline class V1 {
+  final int it;
+  V1(this.it);
+}
```

Comment:
It's a stronger test if we use `final int? it;`, because this would allow a variable of type `V1` to have the value null, but it should _still_ be a compile-time error to use an implicit initialization to null to have that effect.

Reply:
Thanks!

Reply:
Thanks!

It looks funny that the error is reported at `print(x)`, but this is of course because the variable could be initialized at several different locations (`var x; if (blah) x = 1; else x = 2;`), which means that it is only known for sure that something is wrong at the point where the variable is evaluated, and it isn't 'definitely assigned'.

I think it would be really helpful to add a few words about this in a comment.